src/librustc_errors/emitter.rs

```diff
@@ -31,6 +31,11 @@ const ANONYMIZED_LINE_NUM: &str = "LL";
 pub trait Emitter {
     /// Emit a structured diagnostic.
     fn emit(&mut self, db: &DiagnosticBuilder);
+
+    /// Check if should show explanations about "rustc --explain"
+    fn should_show_explain(&self) -> bool {
```

Comment:
[Earlier criticism](https://github.com/rust-lang/rust/pull/48684#discussion_r172004263) of changing the contract of `EmitterWriter` to explicitly reference the `--explain` functionality still applies.